src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarMain.test.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount } from 'enzyme';
+
+import BrowserNavBarMain, {
+  Content as BrowserNavBarMainContent
+} from './BrowserNavBarMain';
+
+import ChromosomeNavigator from 'src/content/app/browser/chromosome-navigator/ChromosomeNavigator';
+import BrowserNavBarRegionSwitcher from './BrowserNavBarRegionSwitcher';
+
+jest.mock(
+  'src/content/app/browser/chromosome-navigator/ChromosomeNavigator',
+  () => () => <div>ChromosomeNavigator</div>
+);
+jest.mock('./BrowserNavBarRegionSwitcher', () => () => (
+  <div>BrowserNavBarRegionSwitcher</div>
+));
+
+describe('BrowserNavBarMain', () => {
+  describe('rendering', () => {
+    it('renders without errors', () => {
+      expect(() => mount(<BrowserNavBarMain />)).not.toThrow();
+    });
+  });
+
+  describe('behaviour', () => {
+    it('renders chromosome visualization by default', () => {
+      const wrapper = mount(<BrowserNavBarMain />);
+      expect(wrapper.find(ChromosomeNavigator).length).toBe(1);
+      expect(wrapper.find(BrowserNavBarRegionSwitcher).length).toBe(0);
+    });
+
+    it('renders RegionSwitcher when user clicks on Change', () => {
+      const wrapper = mount(<BrowserNavBarMain />);
+      const changeButton = wrapper.find('.contentSwitcher');
+      act(() => {
+        changeButton.simulate('click');
+      });
+      wrapper.update();
```

Comment:
Here and below, why do we need `act` and `wrapper.update` ?

Reply:
A click on the changeButton will update component's state.

When react got updated to version 16.8, it actively complained when code causing state updates was not wrapped in an `act`, warning that `When testing, code that causes React state updates should be wrapped into act(...)` (example: `https://github.com/facebook/react/issues/14769`).

I have tried to remove both `act` and `wrapper.update`, and then re-ran the test. It passed, and I could see by using logging that react component managed to get correctly re-rendered. Investigating the reason for this, I found that Enzyme now wraps all calls to `simulate` in an `act`. It wasn't doing so when we started testing 🤷‍♂

Reply:
✔️